docs/configuration/mongodb-event-flattening.asciidoc

```diff
@@ -67,6 +67,64 @@ transforms=unwrap,...
 transforms.unwrap.type=io.debezium.connector.mongodb.transforms.UnwrapFromMongoDbEnvelope
 ----
 
+=== Array encoding
+
+The SMT converts MongoDB arrays into arrays as defined by Apache Connect (or Apache Avro) schema.
+The problem is that such arrays must contains elements of the same time.
+MongoDB allows the user to store elements of heterogeneous types into the same array.
+To bypass this impedance mismatch it is possible to encode the array in two different ways using `array.encoding` configuration option.
+
+[source]
+----
+transforms=unwrap,...
+transforms.unwrap.type=io.debezium.connector.mongodb.transforms.UnwrapFromMongoDbEnvelope
+transforms.unwrap.array.encoding=<array|struct>
+----
+
+Value `array` (the default) will encode arrays as the array datatype.
+It is user's responsibility to ensure that all elements for a given array instance are of the same time.
+This option is a restricting one but offers easy processing of arrays by downstream clients.
+
+Value `struct` will convert the array into a *struct* of *structs* in the similar way as done by http://bsonspec.org/[BSON serialization].
+The main *struct* contains fields named `_0`, `_1`, `_2` etc. where the name represents the index of the element in the array.
+Every element is then passed as the value for the give field.
+
+Let's suppose an example source MongoDB document with array with heterogeneous types
+[source,json,indent=0]
+----
+{
+    "_id": 1,
+    "a1": [
```

Comment:
Why isn't this simply `"a1" : [ 1, "c" ]`?

Reply:
I wanted to demonstrate that i works not only for simple types but for documents too.